cmd/gen-key/ecdsa.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/miekg/pkcs11"
+)
+
+var stringToCurve = map[string]elliptic.Curve{
+	"P-224": elliptic.P224(),
+	"P-256": elliptic.P256(),
+	"P-384": elliptic.P384(),
+	"P-521": elliptic.P521(),
+}
+
+var curveToOID = map[elliptic.Curve][]byte{
+	elliptic.P224(): []byte{6, 5, 43, 129, 4, 0, 33},
+	elliptic.P256(): []byte{6, 8, 42, 134, 72, 206, 61, 3, 1, 7},
+	elliptic.P384(): []byte{6, 5, 43, 129, 4, 0, 34},
+	elliptic.P521(): []byte{6, 5, 43, 129, 4, 0, 35},
+}
+
+func ecArgs(label string, curve elliptic.Curve, compatMode bool) ([]*pkcs11.Mechanism, []*pkcs11.Attribute, []*pkcs11.Attribute) {
+	encodedCurve := curveToOID[curve]
+	log.Printf("\tEncoded curve parameters for %s: %X\n", curve.Params().Name, encodedCurve)
```

Comment:
This is the DER encoding of the OID, this is so that we don't have to call `asn1.Marshal` for something we already know the encoded value of.

Reply:
Ah, good point. I missed that since the initialization format (`[]byte{6, 8, 42, 134, 72, 206, 61, 3, 1, 7},`) looks a lot like an OID. Maybe you could make this clearer by calling the map `curveToOIDDER`?